api/proto/teleport/usageevents/v1/usageevents.proto

```diff
@@ -147,6 +147,91 @@ message UIDiscoverResourceSelectionEvent {
+// For SSH this is the Teleport 'install-node' script.
+//
+// For Kubernetes this is the teleport-agent helm chart installation.
+//
+// For Database Access this step is the installation of the teleport 'install-db' script.
+// It can be skipped if the cluster already has a Database Service capable of proxying the database.
+message UIDiscoverDeployServiceEvent {
+  DiscoverMetadata metadata = 1;
+  DiscoverResourceMetadata resource = 2;
+  DiscoverStepStatus status = 3;
+}
+
+// UIDiscoverDatabaseRegisterEvent is emitted when a user registers a database resource
+// and goes to the next step.
+message UIDiscoverDatabaseRegisterEvent {
+  DiscoverMetadata metadata = 1;
+  DiscoverResourceMetadata resource = 2;
+  DiscoverStepStatus status = 3;
+}
+
+// UIDiscoverDatabaseConfigureMTLSEvent is emitted when a user configures mutual TLS for self-hosted database
+// and goes to the next step.
+message UIDiscoverDatabaseConfigureMTLSEvent {
+  DiscoverMetadata metadata = 1;
+  DiscoverResourceMetadata resource = 2;
+  DiscoverStepStatus status = 3;
+}
+
+// UIDiscoverDesktopActiveDirectoryToolsInstallEvent is emitted when the user is asked to run the install Active Directory tools script.
+// This happens on the Desktop flow.
+message UIDiscoverDesktopActiveDirectoryToolsInstallEvent {
+  DiscoverMetadata metadata = 1;
+  DiscoverResourceMetadata resource = 2;
+  DiscoverStepStatus status = 3;
+}
+
+// UIDiscoverDesktopActiveDirectoryConfigureEvent is emitted when the user is asked to run the Configure Active Directory script.
+// This happens on the Desktop flow.
+message UIDiscoverDesktopActiveDirectoryConfigureEvent {
+  DiscoverMetadata metadata = 1;
+  DiscoverResourceMetadata resource = 2;
+  DiscoverStepStatus status = 3;
+}
+
+// UIDiscoverAutoDiscoveredResourcesEvent is emitted when the user is presented with the list of auto discovered resources.
+// resources_count field must contain the number of discovered resources when the user leaves the screen.
+message UIDiscoverAutoDiscoveredResourcesEvent {
+  DiscoverMetadata metadata = 1;
+  DiscoverResourceMetadata resource = 2;
+  DiscoverStepStatus status = 3;
+  int64 resources_count = 4;
+}
+
+// UIDiscoverDatabaseConfigureIAMPolicyEvent is emitted when a user configured IAM for RDS database
+// and proceeded to the next step.
+message UIDiscoverDatabaseConfigureIAMPolicyEvent {
+  DiscoverMetadata metadata = 1;
+  DiscoverResourceMetadata resource = 2;
+  DiscoverStepStatus status = 3;
+}
+
+// UIDiscoverPrincipalsConfigureEvent emitted on "Setup Access" screen when user has updated their principals
+// and proceeded to the next step.
+message UIDiscoverPrincipalsConfigureEvent {
+  DiscoverMetadata metadata = 1;
+  DiscoverResourceMetadata resource = 2;
+  DiscoverStepStatus status = 3;
+}
+
+// UIDiscoverTestConnectionEvent emitted on the "Test Connection" screen
+// when the user clicked tested connection to their resource.
```

Comment:
hm, i think i did this one wrong. all of our success events happen after user clicks on `next step`. this is what currently happens on test connection:

1) user clicks `test connection` and it fails, emit `test connection failed` event with error traces (we also send error event on non-connection error like network failure)
2) user clicks on `next step` and that's when we send of `test connection success` event, but i can see how this can be a false positive since it isn't required of user to test the connection to go to next step (finish step)

if we want to send a success event after they get successful testing:
- should i be sending a `success event` on the very first successful testing? (i'm thinking this is enough)
- or per successful testing? (they can test out different combos)
- or send a success event only if after they `click next` and the last testing was successful (or if they had at least ONE success among their testing out multiple cases).

Reply:
IMO, we should send an event every time they test the connection.
If they don't test the connection and hit `next step` we could either:
- don't send any test connection event
- send a test connection event with the skipped step status

Our current approach also works, because that should be what happens 99% of the time.
(that percentage is totally made up by me, so it might be far from reality :sweat_smile: )